File: Question2/Question2.js
'use strict';

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

function solution(files) {
    const freq = {};
    const len = files.length;

    for (let i = 0; i < len; i++) {
        const char = files[i];

        freq[char] = freq[char] ? freq[char] + 1 : 1;
    }

    let palindrome_length = 0;
    let has_odd = false;

    for (const char in freq) {
        const count = freq[char];

        if (count % 2 === 0) palindrome_length += count;
        else {
            palindrome_length += count - 1;
            has_odd = true;
        }
    }

    if (has_odd) palindrome_length++;

    return palindrome_length;
}

readline.question('', files => {
    const answer = solution(files);
    console.log(answer);

    readline.close();
});

File: Question2/Question2.py
def solution(files):
    freq = [0] * 128

    for c in files: freq[ord(c)] += 1

    palindrome_length = 0

    for i in freq:
        palindrome_length += i // 2 * 2
        if palindrome_length % 2 == 0 and i % 2 == 1:
            palindrome_length += 1

    return palindrome_length

def main():
    str = input()
    answer = solution(str)
    print(answer)

if __name__ == "__main__":
    main()
